pkg/reconciler/ingress/reconciler_accessors.go

```diff
@@ -0,0 +1,122 @@
+/*
+Copyright 2019 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package ingress
+
+import (
+	"github.com/knative/serving/pkg/apis/networking/v1alpha1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes"
+	corev1listers "k8s.io/client-go/listers/core/v1"
+	"k8s.io/client-go/tools/record"
+	sharedclientset "knative.dev/pkg/client/clientset/versioned"
+	istiolisters "knative.dev/pkg/client/listers/istio/v1alpha3"
+	"knative.dev/pkg/tracker"
+)
+
+// ingressFinalizer is the name that we put into the resource finalizer list, e.g.
+//  metadata:
+//    finalizers:
+//    - ingresses.networking.internal.knative.dev
+var (
+	ingressResource  = v1alpha1.Resource("ingresses")
+	ingressFinalizer = ingressResource.String()
+)
+
+// ReconcilerAccessor defines functions that access reconciler data specific to Ingress types
+type ReconcilerAccessor interface {
+	DeepCopy(v1alpha1.IngressAccessor) v1alpha1.IngressAccessor
+
+	GetFinalizer() string
```

Comment:
Could this just be passed to the base reconciler constructor?

Reply:
Done.  Ty!